### PR TITLE
Filters API polish

### PR DESCRIPTION
**Pull request by ranquild**
Closes https://github.com/metabase/metabase/issues/34190

Changes:
- Universally rely on `filterableColumnOperators` to compute the list of operators
- Operator-based functions now accept opaque `FilterOperator`
- For column-based temporal buckets, accept opaque `Bucket`
- Implement specific & relative & exclude date functions properly
- Implement coordinate filters and handle `inside`

### CODE REVIEW

File: frontend/src/metabase-lib/types.ts [hunk 1]
```diff
@@ -211,10 +218,21 @@ export type FilterParts =
   | ExcludeDateFilterParts
   | TimeFilterParts;
 
-export type StringFilterOperatorName = typeof STRING_FILTER_OPERATORS[number];
+export type DateParts = {
```
Comment (ranquild) [inline frontend/src/metabase-lib/types.ts:221]: These are designed to be consumed by `moment`

File: frontend/src/metabase/common/components/FilterPicker/BooleanFilterPicker.tsx [hunk 2]
```diff
@@ -7,40 +7,15 @@ import { Header } from "./Header";
...
 
 type Option = {
   name: string;
   type: OptionType;
-  operator: Lib.FilterOperatorName;
+  operator: Lib.FilterOperator;
   isAdvanced?: boolean;
 };
 
-const OPTIONS: Option[] = [
```
Comment (ranquild) [inline frontend/src/metabase/common/components/FilterPicker/BooleanFilterPicker.tsx:19]: Rely on `filterableColumnOperators`

File: frontend/src/metabase-lib/types.ts [hunk 3]
```diff
@@ -224,67 +242,62 @@ export type StringFilterOptions = {
...
+  operator: FilterOperator;
   column: ColumnMetadata;
   values: number[];
 };
 
-export type BooleanFilterOperatorName = typeof BOOLEAN_FILTER_OPERATORS[number];
+export type CoordinateFilterParts = {
+  operator: FilterOperator;
+  column: ColumnMetadata;
+  longitudeColumn?: ColumnMetadata;
```
Comment (ranquild) [inline frontend/src/metabase-lib/types.ts:254]: btw, the logic is that if there is just one longitude column in the data source, it's omitted (the column is duplicated in mbql). But if there are multiple ones you can select which one should be used

Comment (kulyk) [reply frontend/src/metabase-lib/types.ts:254]: Should we maybe add a comment about it somewhere? The way it works isn't really straightforward

**Review by kulyk — APPROVED** [review]

**Pull request merged**